src/infrastructure/httpServer/HttpServer.ts

```diff
@@ -203,6 +190,40 @@ export class HttpServer extends ConnectorInfrastructure<HttpServerConfiguration>
         this.app.use(genericErrorHandler(this.connectorMode));
     }
 
+    private useApiKey() {
+        if (!this.configuration.apiKey) {
+            switch (this.connectorMode) {
+                case "debug":
+                    return;
+                case "production":
+                    throw new Error("No API key set in configuration. This is required in production mode.");
+            }
+        }
+
+        const apiKeyStrength = checkAPIKeyStrength(this.configuration.apiKey, [
+            { id: 0, value: "Very Weak", minDiversity: 0, minLength: 0 },
+            { id: 1, value: "Strong", minDiversity: 0, minLength: 30 }
+        ]);
+        if (apiKeyStrength.value !== "Strong") {
+            this.logger.warn(
+                "The configured API key does not meet the requirements. It must be 30 chars and contain at least 1 digit, 1 uppercase, 1 lowercase, and 1 special char."
+            );
+        }
+
+        this.app.use(async (req, res, next) => {
+            const apiKeyFromHeader = req.headers["x-api-key"];
+            if (!apiKeyFromHeader || apiKeyFromHeader !== this.configuration.apiKey) {
+                // sleep for 1 to 4 seconds
```

Comment:
Remove comment please.

Reply:
I have added this on purpose b/c I was not sure if the 1-4 seconds logic is understandable without. I'm fine with removing it if everyone understands it.

Reply:
I think this line is easily looked-up in case someone is interested in the time of sleep. IMO our guideline to not use comments is more important here.